### PR TITLE
[docs] Add a note on Native AOT diagnostics support on iOS-like platforms

### PR DESCRIPTION
**Pull request by ivanpovazan**
## Summary

This PR adds a note on Native AOT diagnostics support when targeting iOS-like platforms which includes a link to the iOS-specific documentation.

--- 
Contributes to: https://github.com/dotnet/runtime/issues/106799

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/deploying/native-aot/diagnostics.md](https://github.com/dotnet/docs/blob/2986ff3e5d7c5dd1ee5dc427713a1c6ac37eb249/docs/core/deploying/native-aot/diagnostics.md) | [Diagnostics and instrumentation](https://review.learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/diagnostics?branch=pr-en-us-43399) |

### CODE REVIEW

File: docs/core/deploying/native-aot/diagnostics/desktop-platforms.md
Comment (jkotas) [inline docs/core/deploying/native-aot/diagnostics/desktop-platforms.md:3]: "desktop platforms" is our internal term with very specific definition. It is not a term that our users understand.

For example, it is very odd to call Linux containers as desktop platform.

**Comment by ivanpovazan** [comment]
@jkotas thank you, I made a note that there is a difference in diagnostics support and add a placeholder URL which will point to the iOS-specific documentation added to MAUI docs: https://github.com/dotnet/docs-maui/pull/2614
Will follow-up on this and correct the URL once the mentioned PR gets merged in.

File: docs/core/deploying/native-aot/diagnostics.md
Comment (jkotas) [inline docs/core/deploying/native-aot/diagnostics.md:16]: I think it would be more consistent with our documentation style to add this link in "See also" section at the end of the doc. Random example from the existing docs https://learn.microsoft.com/en-us/dotnet/core/deploying/trimming/trim-self-contained#see-also.

Majority of this doc should be applicable to iOS too. The only wrinkle that I can see is that this doc talks explicitly about CoreCLR in several places. You can fix those by saying e.g. "full .NET runtime" so that it applies to Mono as well.

Comment (jkotas) [reply docs/core/deploying/native-aot/diagnostics.md:16]: Also, left comment on your other PR at https://github.com/dotnet/docs-maui/pull/2614#discussion_r1838437503

Comment (ivanpovazan) [reply docs/core/deploying/native-aot/diagnostics.md]: Fixed.

**Review by jkotas — APPROVED** [review]

**Comment by ivanpovazan** [comment]
Can we merge this or we need an approval from someone from @dotnet/docs ? 
Thanks.

**Comment by gewarren** [comment]
> The usual operating procedure is to wait for signoff from the docs team.

I'll take a look today.

**Review by rolfbjarne — APPROVED** [review]

File: docs/core/deploying/native-aot/diagnostics.md
Comment (gewarren) [inline docs/core/deploying/native-aot/diagnostics.md:84]:
```suggestion
- [Native AOT diagnostic support on iOS and Mac Catalyst](/dotnet/maui/deployment/nativeaot#native-aot-diagnostic-support-on-ios-and-mac-catalyst)
```

File: docs/core/deploying/native-aot/diagnostics.md
Comment (gewarren) [inline docs/core/deploying/native-aot/diagnostics.md:11]:
```suggestion
Native AOT shares some, but not all, diagnostics and instrumentation capabilities with the full .NET runtime. Apps that are [trim-compatible](../trimming/prepare-libraries-for-trimming.md) shouldn't have behavioral differences, so investigations often apply to both runtimes. As such, it's sometimes appropriate to diagnose and debug problems in the full .NET runtime, as it has a rich selection of available diagnostic utilities. Nonetheless, some information can only be gathered after publishing, so Native AOT also provides post-publish diagnostic tooling.
```

Comment (gewarren) [reply docs/core/deploying/native-aot/diagnostics.md:11]: I did this to shorten the sentence length (better for machine translation).

**Review by gewarren — APPROVED** [review]

File: docs/core/deploying/native-aot/diagnostics.md
Comment (ivanpovazan) [reply docs/core/deploying/native-aot/diagnostics.md:84]: @gewarren the URL needs to point to the .NET 9 version of the documentation. Not sure how this would work with relative links?

Comment (gewarren) [reply docs/core/deploying/native-aot/diagnostics.md:84]: It's better not to pin it to a version, so when the next version comes out, it's not pinned to an old version. Also, I tried out the link I suggested, and it defaults to .NET 9, so I think it's fine as suggested.

https://learn.microsoft.com/dotnet/maui/deployment/nativeaot#native-aot-diagnostic-support-on-ios-and-mac-catalyst

Comment (ivanpovazan) [reply docs/core/deploying/native-aot/diagnostics.md]: Thanks for the clarification, I added your suggestion. 
I think we are now good to merge this?

**Pull request merged**
